src/components/Post/GovernanceSideBar/Referenda/VoteReferendum.tsx

```diff
@@ -342,62 +498,160 @@ const VoteReferendum = ({ className, referendumId, onAccountChange, lastVote, se
+								{
+									(window as any).walletExtension?.isNovaWallet && availableWallets[Wallet.NOVAWALLET] &&
+                    <WalletButton disabled={!apiReady} className={`${wallet === Wallet.NOVAWALLET? 'border border-solid border-pink_primary  w-[64px] h-[48px]': 'w-[64px] h-[48px]'}`} onClick={(event) => handleWalletClick((event as any), Wallet.NOVAWALLET)} name="Nova Wallet" icon={<WalletIcon which={Wallet.NOVAWALLET} className='h-6 w-6' />} />
+								}
+								{
+									['polymesh'].includes(network) && availableWallets[Wallet.POLYWALLET]?
+										<WalletButton disabled={!apiReady} onClick={(event) => handleWalletClick((event as any), Wallet.POLYWALLET)} className={`${wallet === Wallet.POLYWALLET? 'border border-solid border-pink_primary  w-[64px] h-[48px]': 'w-[64px] h-[48px]'}`}  name="PolyWallet" icon={<WalletIcon which={Wallet.POLYWALLET} className='h-6 w-6'  />} />
+										: null
+								}
+							</div>
+							{balanceErr.length > 0 && <Alert type='info' message={balanceErr} showIcon className='mb-4'/>}
+							{walletErr.error === 1 && !loadingStatus.isLoading && <Alert message={walletErr.message} description={walletErr.description} showIcon/>}
+							{accounts.length === 0  && wallet && !loadingStatus.isLoading && <Alert message='No addresses found in the address selection tab.' showIcon type='info' />}
 							{
-								(window as any).walletExtension?.isNovaWallet && availableWallets[Wallet.NOVAWALLET] &&
-                    <WalletButton disabled={!apiReady} className={`${wallet === Wallet.POLYWALLET? 'border border-solid border-pink_primary': ''}`} onClick={(event) => handleWalletClick((event as any), Wallet.NOVAWALLET)} name="Nova Wallet" icon={<WalletIcon which={Wallet.NOVAWALLET} className='h-6 w-6' />} />
+								accounts.length > 0 ?
+									<AccountSelectionForm
+										title='Vote with Account'
+										accounts={accounts}
+										address={address}
+										withBalance
+										onAccountChange={onAccountChange}
+										onBalanceChange={handleOnBalanceChange}
+										className={`${poppins.variable} ${poppins.className} text-sm font-normal text-[#485F7D]`}
+										inputClassName='rounded-[4px] px-3 py-1'
+										withoutInfo={true}
+									/>
+									: walletErr.message.length === 0 && !wallet && !loadingStatus.isLoading ? <Alert message='Please select a wallet.' showIcon type='info' />: null
 							}
+
+							{/* aye nye split abstain buttons */}
+							<h3 className='inner-headings mt-[24px] mb-[2px]'>Choose your vote</h3>
+							<Segmented
+								block
+								className={`${className} mb-6 border-solid border-[1px] bg-white border-[#D2D8E0] rounded-[4px] w-full`}
+								size="large"
+								value={vote}
+								onChange={(value) => {
```

Comment:
reset aye and nay vote vars to ZERO_BN onChange here

Reply:
this is done